src/simulation/hdf_simulation_store.jl

```diff
@@ -341,11 +365,26 @@ function read_result(
     index::Union{DecisionModelIndexType, EmulationModelIndexType},
 )
     data, columns = _read_data_columns(store, model_name, key, index)
+    return _make_dataframe(data, columns)
+end
 
-    if (ndims(data) < 2 || size(data)[1] == 1) && size(data)[2] != size(columns)[1]
-        data = reshape(data, length(data), 1)
-    end
-    return DataFrames.DataFrame(data, columns; copycols = false)
+function _make_denseaxisarray(
+    data::Matrix{Float64},
+    columns::Tuple{Vector{String}},
+)
+    return DenseAxisArray(permutedims(data), columns[1], 1:size(data)[1])
```

Comment:
Should there be a check if the length of columns is greater than 1?

Reply:
This is already blocked by the type `Tuple{Vector{String}}` can only be indexed by `[1]`